sdk/core/azure-core/inc/azure/core/internal/http/user_agent.hpp

```diff
@@ -11,12 +11,25 @@
 #include <string>
 
 namespace Azure { namespace Core { namespace Http { namespace _detail {
+  // NOTE: Treat Azure::Core::Http::_detail::UserAgentGenerator::GenerateUserAgent() as _internal -
+  // it is being/has been used by eventhubs.
```

Comment:
Thinking beyond the release issue today (i.e. for a future change), does it make sense to move such a helper class to internal, if we expect SDK clients to use it?

Reply:
Yes, we should effectively create an alias in the `_internal` namespace, and use it in the new version of eventhubs. But we probably still have to keep this one in `_detail` forever, for the compatibility with previously released eventhubs.

Reply:
#6064

Reply:
Since we haven't GA'd evenhubs, we shouldn't bend over backwards to maintain beta to beta compatibility for this "bug" and fix it a cleanly as possible.

Reply:
It would look as if it is Core that broke the no-breaking-change contract, not the eventhubs.

Reply:
Tbh, other SDK packages or end users using something in _detail isn't part of the breaking change contract, so it is safe for us to fix this "correctly" rather than introducing complexity in core, for perpetuity, for no benefit to the end user.

This falls in the camp of a bug fix.

Reply:
I see, using a function from `_detail` could be considered a bug and therefore rightfully fail.